src/lib/optimizer/optimizer.cpp

```diff
@@ -91,15 +88,19 @@ std::shared_ptr<Optimizer> Optimizer::create_default_optimizer() {
 
   optimizer->add_rule(std::make_unique<ExpressionReductionRule>());
 
-  optimizer->add_rule(std::make_unique<PredicateSplitUpRule>());
-
   optimizer->add_rule(std::make_unique<ColumnPruningRule>());
 
   optimizer->add_rule(std::make_unique<ChunkPruningRule>());
 
-  // Run before SubqueryToJoinRule, since the Semi/Anti Joins it introduces are opaque to the JoinOrderingRule
+  // Run once before the JoinOrderingRule without disjunction split-up because the JoinOrderingRule cannot handle UNION
+  // (#1829)
+  optimizer->add_rule(std::make_unique<PredicateSplitUpRule>(false));
+
+  // Run before SubqueryToJoinRule since the Semi/Anti Joins it introduces are opaque to the JoinOrderingRule
   optimizer->add_rule(std::make_unique<JoinOrderingRule>());
 
+  optimizer->add_rule(std::make_unique<PredicateSplitUpRule>());
```

Comment:
Not a full review, but did you test different positions for this? For #1819, it would be helpful if this rule was behind the PredicatePlacementRule. Would that break anything on your side?

Reply:
That works. Nothing breaks and the performance does not notably change.